JotunnLib/Managers/DungeonManager.cs

```diff
@@ -0,0 +1,372 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HarmonyLib;
+using Jotunn.Entities;
+using Jotunn.Managers.MockSystem;
+using Jotunn.Utils;
+using UnityEngine;
+
+namespace Jotunn.Managers
+{
+    /// <summary>
+    ///    Manager for handling all custom data added to the game related to creatures.
+    /// </summary>
+    public class DungeonManager : IManager
+    {
+        private static DungeonManager _instance;
+
+        /// <summary>
+        ///     The singleton instance of this manager.
+        /// </summary>
+        public static DungeonManager Instance => _instance ??= new DungeonManager();
+
+        /// <summary>
+        ///     Event that gets fired after the vanilla <see cref="DungeonDB"/> has loaded rooms.  Your code will execute
+        ///     every time a main scene is started (on joining a game). <br />
+        ///     If you want to execute just once you will need to unregister from the event after execution.
+        /// </summary>
+        public static event Action OnVanillaRoomsAvailable;
+        
+        public static event Action OnRoomsRegistered;
```

Comment:
the doc string is missing

Reply:
Addressed in https://github.com/Valheim-Modding/Jotunn/pull/434/commits/98905482002e29a1e4d24363206a1298f9ffeeec